integration_tests/__tests__/test_environment_async.test.js

```diff
@@ -0,0 +1,22 @@
+/**
+ * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+'use strict';
+
+const skipOnWindows = require('../../scripts/skip_on_windows');
+const runJest = require('../runJest');
+const fs = require('fs');
+
+skipOnWindows.suite();
+
+it('triggers setup/teardown hooks', () => {
+  fs.writeFileSync('/tmp/teardown', '');
+  runJest('test-environment-async');
+  const teardown = fs.readFileSync('/tmp/teardown', 'utf8');
```

Comment:
You can use `writeFiles` and `cleanup` helpers from `integration_tests/utils.js` 🙂